File: flutter_app/lib/features/productos/domain/usecases/get_producto.dart
import '../../../../core/usecase/usecase.dart';
import '../entities/producto.dart';
import '../repositories/producto_repository.dart';

/// Obtiene el detalle de un producto por su id.
class GetProducto implements UseCase<Producto, int> {
  final ProductoRepository repository;
  GetProducto(this.repository);

  @override
  Future<Producto> call(int id) => repository.getProducto(id);
}

File: react_native_app/src/features/productos/domain/usecases/getProducto.ts
import { UseCase } from '../../../../core/usecase/usecase';
import { Producto } from '../entities/Producto';
import { ProductoRepository } from '../repositories/ProductoRepository';

/** Obtiene el detalle de un producto por su id. */
export class GetProducto implements UseCase<Producto, number> {
  constructor(private readonly repository: ProductoRepository) {}

  call(id: number): Promise<Producto> {
    return this.repository.getProducto(id);
  }
}
